File: java/src/main/java/frc/robot/subsystems/GatheringSubsystem.java

package frc.robot.subsystems;


import edu.wpi.first.wpilibj.PWMVictorSPX;
import edu.wpi.first.wpilibj.Servo;
import edu.wpi.first.wpilibj2.command.SubsystemBase;

import static frc.robot.Constants.GC;

public class GatheringSubsystem extends SubsystemBase {

    private final PWMVictorSPX motor;
    private final Servo servoLeft;
    private final Servo servoRight;

    public GatheringSubsystem() {
        motor = new PWMVictorSPX(GC.MOTOR_GATHERING);
        servoLeft = new Servo(GC.SERVO_PORTS[0]);
        servoRight = new Servo(GC.SERVO_PORTS[1]);
    }

    public void takeIn() {
        motor.set(-0.7);
    }

    public void takeOut() {
        motor.set(0.7);
    }

    public void stop() {
        motor.set(0.0);
    }

    public void stopServos() {
        servoLeft.stopMotor();
        servoRight.stopMotor();
    }

    public void pressServos() {
        servoLeft.set(0.7);
        servoRight.set(0.65);
    }

    public void releaseServos() {
        servoLeft.set(0.1);
        servoRight.set(0.1);
    }
}

File: c++/src/main/cpp/subsystems/GatheringSubsystem.cpp
/*----------------------------------------------------------------------------*/
/* Copyright (c) 2019 FIRST. All Rights Reserved.                             */
/* Open Source Software - may be modified and shared by FRC teams. The code   */
/* must be accompanied by the FIRST BSD license file in the root directory of */
/* the project.                                                               */
/*----------------------------------------------------------------------------*/

#include "subsystems/GatheringSubsystem.h"

GatheringSubsystem::GatheringSubsystem()
	: m_motor{gc::kMotorGathering},
	  m_servoLeft{gc::kServoPorts[0]},
	  m_servoRight{gc::kServoPorts[1]}
{
}

void GatheringSubsystem::TakeIn()
{
	m_motor.Set(-0.6);
}

void GatheringSubsystem::TakeOut()
{
	m_motor.Set(0.6);
}

void GatheringSubsystem::Stop()
{
	m_motor.Set(0.0);
}

void GatheringSubsystem::StopServos()
{
	m_servoLeft.StopMotor();
	m_servoRight.StopMotor();
}

void GatheringSubsystem::PressServos()
{
	m_servoLeft.Set(0.7);
	m_servoRight.Set(1.1);
}

void GatheringSubsystem::ReleaseServos()
{
	m_servoLeft.Set(0.1);
	m_servoRight.Set(0.4);
}
